client/ayon_core/hosts/substancepainter/plugins/load/load_mesh.py

```diff
@@ -24,49 +151,37 @@ class SubstanceLoadProjectMesh(load.LoaderPlugin):
     order = -10
     icon = "code-fork"
     color = "orange"
+    project_templates = []
 
-    options = [
-        qargparse.Boolean(
-            "preserve_strokes",
-            default=True,
-            help="Preserve strokes positions on mesh.\n"
-                 "(only relevant when loading into existing project)"
-        ),
-        qargparse.Boolean(
-            "import_cameras",
-            default=True,
-            help="Import cameras from the mesh file."
-        )
-    ]
-
-    def load(self, context, name, namespace, data):
+    def load(self, context, name, namespace, options=None):
 
         # Get user inputs
-        import_cameras = data.get("import_cameras", True)
-        preserve_strokes = data.get("preserve_strokes", True)
+        result = SubstanceProjectConfigurationWindow.prompt(
+            self.project_templates)
+        template_name = result.template_name
+        if template_name is None:
+            return
+        import_cameras = result.import_cameras
+        template = get_template_by_name(template_name, self.project_templates)
```

Comment:
I think getting the template resulting values are best to get from the dialog directly - so that the dialog can return the resulting values including any overrides to the import cameras, etc.

```python
template = dialog.get_template()
if not template:
    # user cancelled
    return

...
```